src/persist/src/indexed/encoding.rs

```diff
@@ -211,6 +212,28 @@ impl BlobTraceUpdates {
             BlobTraceUpdates::Both(_, structured) => Some(structured),
         }
     }
+
+    /// Return the [`ColumnarRecordsStructuredExt`] of the blob.
+    pub fn get_or_make_structured<K: Codec, V: Codec>(
+        &mut self,
+        key_schema: &K::Schema,
+        value_schema: &V::Schema,
+    ) -> &ColumnarRecordsStructuredExt {
+        match self {
+            BlobTraceUpdates::Row(records) => {
+                let key = codec_to_schema2::<K>(key_schema, records.keys()).expect("valid keys");
+                let val =
+                    codec_to_schema2::<V>(value_schema, records.vals()).expect("valid values");
+                *self = BlobTraceUpdates::Both(
+                    records.clone(),
+                    ColumnarRecordsStructuredExt { key, val },
+                );
+                // Recurse at most once, since this data is now structured.
```

Comment:
I have an irrational fondness for this technique

Reply:
😅